bgzip.c

```diff
@@ -103,6 +106,83 @@ static int confirm_filename(int *is_forced, const char *name, const char *ext)
     return ask_yn();
 }
 
+/* getfilespec - get file status data
+   path        - file path for which status to be retrieved
+   filestatus  - pointer to status structure in which the data to be stored
+   returns 0 on success and -1 on failure
+*/
+static int getfilespec(const char *path, void *filestatus)
+{
+#ifdef _WIN32
+    struct _stat *status = (struct _stat*)filestatus;
+#else
+    struct stat *status = (struct stat*)filestatus;
+#endif //_WIN32
```

Comment:
This shouldn't be necessary.

We use MinGW which supports this, but even MSVC appears to have "struct stat" as a native type.  Eg see a draft MSVC native port here: https://github.com/refresh-bio-sandbox/htslib-msvc/blob/develop/hts.c#L4600

If we needed to do OS specific shenanigans, it's better hidden away in os/* and hts_os.c.

Reply:
Not expecting this API to be used anywhere else and is for internal use only, so not adding to hts_os.
updated to avoid _stat and uses stat in all cases.
Note: to set time, uses _utime in windows and utimes with micro sec part, if any, set to 0 in linux and Mac.